src/pkg/bundle/deploy.go

```diff
@@ -409,6 +410,21 @@ func addOverrideValue(overrides map[string]map[string]*values.Options, component
 		val := fmt.Sprintf("%s=%s", valuePath, j)
 		overrides[component][chart].JSONValues = append(overrides[component][chart].JSONValues, val)
 	default:
+		// Check for any templated variables if pkgVars set
```

Comment:
1. Optimize Regular Expression Compilation:
Currently, the regular expression is compiled within the executed block of code. This is efficient for occasional use, but if this function is called frequently, it might lead to unnecessary overhead. A common practice is to compile regular expressions once and reuse them throughout the application. This can be achieved by declaring the compiled regex outside of the function scope. This change could improve the performance of your code.

```go
// Consider moving this outside the function to compile it only once
   var templatedVarRegex = regexp.MustCompile(`\${([^}]+)}`)
 ```
 
 2. Enhance Handling of Multiple Templated Variables:
The current implementation replaces the first occurrence of a templated variable. If a string contains multiple templated variables, it would be beneficial to replace all of them. This adjustment ensures that all templated variables are processed, making the function more versatile.

```go
   // Use ReplaceAllStringFunc to handle all occurrences of templated variables
   replacedValue := templatedVarRegex.ReplaceAllStringFunc(templatedVariable, func(match string) string {
       variableName := templatedVarRegex.FindStringSubmatch(match)[1]
       if varValue, ok := pkgVars[variableName]; ok {
           return varValue
       }
       // Consider handling missing variables more explicitly
       return fmt.Sprintf("${%s_not_found}", variableName)
   })
   value = replacedValue
  ```